Add plugin: Kindle Vocab

bao-tg opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/bao-tg/kindle-vocab

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [x]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [ ] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

bao-tg commented:
I have finished all of the required changes. Please check it again.

bao-tg commented:
Updated

bao-tg commented:
Fix all issues

bao-tg commented:
> ['Sync Kindle Vocabulary'](https://github.com/bao-tg/kindle-vocab/blob/49473a602c362e760b7ceccb51342bdace46ab22/src/ribbon/RibbonHandlers.ts#L6), ['Sort Order'](https://github.com/bao-tg/kindle-vocab/blob/cf7445283bf6e0d0030413480bed164e6588f6d6/src/settings/SettingTab.ts#L18), ['Newest First (Timestamp)'](https://github.com/bao-tg/kindle-vocab/blob/cf7445283bf6e0d0030413480bed164e6588f6d6/src/settings/SettingTab.ts#L22), ['Unlearned First'](https://github.com/bao-tg/kindle-vocab/blob/cf7445283bf6e0d0030413480bed164e6588f6d6/src/settings/SettingTab.ts#L23), ['Markdown Output Folder'](https://github.com/bao-tg/kindle-vocab/blob/cf7445283bf6e0d0030413480bed164e6588f6d6/src/settings/SettingTab.ts#L34), ['Assets Folder'](https://github.com/bao-tg/kindle-vocab/blob/cf7445283bf6e0d0030413480bed164e6588f6d6/src/settings/SettingTab.ts#L62), ['Sync Options'](https://github.com/bao-tg/kindle-vocab/blob/cf7445283bf6e0d0030413480bed164e6588f6d6/src/settings/SettingTab.ts#L15), ['Sync Your Vocabulary Builder'](https://github.com/bao-tg/kindle-vocab/blob/cf7445283bf6e0d0030413480bed164e6588f6d6/src/modals/SyncDatabaseModal.ts#L19), ['Upload a Dictionary File (CSV)'](https://github.com/bao-tg/kindle-vocab/blob/cf7445283bf6e0d0030413480bed164e6588f6d6/src/modals/DictionaryUploadModal.ts#L22), ['Upload a Vocabulary File (DB)'](https://github.com/bao-tg/kindle-vocab/blob/cf7445283bf6e0d0030413480bed164e6588f6d6/src/modals/DatabaseUploadModal.ts#L25) [Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)
> 
> [containerEl.createEl('h3', { text: 'Sync Options' });](https://github.com/bao-tg/kindle-vocab/blob/cf7445283bf6e0d0030413480bed164e6588f6d6/src/settings/SettingTab.ts#L15) [Only use headings under settings if you have more than one section.](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)
> 
> [containerEl.createEl('h3', { text: 'Sync Options' });](https://github.com/bao-tg/kindle-vocab/blob/cf7445283bf6e0d0030413480bed164e6588f6d6/src/settings/SettingTab.ts#L15) [Avoid "options" in settings headings](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Avoid+%22settings%22+in+settings+headings)
> 
> [containerEl.createEl('h3', { text: 'Sync Options' });](https://github.com/bao-tg/kindle-vocab/blob/cf7445283bf6e0d0030413480bed164e6588f6d6/src/settings/SettingTab.ts#L15) For section headings in settings use:
> 
> ```ts
> new Setting(containerEl).setName('name here').setHeading();
> ```
> 
> [const folderExists = this.app.vault.getAllLoadedFiles()](https://github.com/bao-tg/kindle-vocab/blob/cf7445283bf6e0d0030413480bed164e6588f6d6/src/settings/SettingTab.ts#L46-L47), [const folderExists = this.app.vault.getAllLoadedFiles()](https://github.com/bao-tg/kindle-vocab/blob/cf7445283bf6e0d0030413480bed164e6588f6d6/src/settings/SettingTab.ts#L73-L74) [Avoid iterating all files/folders to find a file/folder by its path](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Avoid+iterating+all+files+to+find+a+file+by+its+path)
> 
> Once you've made the requested changes can you please create a new release so I can more easily test?

Sir, I have fixed all of the requirements requested by you, thank you so much for your insightful suggestions. I've also updated new release https://github.com/bao-tg/kindle-vocab/releases/tag/1.0.2 that is ready to be tested.

Zachatoo commented:
['Sync Your Vocabulary Builder'](https://github.com/bao-tg/kindle-vocab/blob/cf7445283bf6e0d0030413480bed164e6588f6d6/src/modals/SyncDatabaseModal.ts#L19)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

Can you please also set a name for each of your settings? They look out of place with just a description and no name.

---

These are the final changes I'm requesting, apologies for the back and forth.

Once you've made these changes remember to make a new release!

bao-tg commented:
> ['Sync Your Vocabulary Builder'](https://github.com/bao-tg/kindle-vocab/blob/cf7445283bf6e0d0030413480bed164e6588f6d6/src/modals/SyncDatabaseModal.ts#L19) [Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)
> 
> Can you please also set a name for each of your settings? They look out of place with just a description and no name.
> 
> These are the final changes I'm requesting, apologies for the back and forth.
> 
> Once you've made these changes remember to make a new release!

I've fixed all the UI (use sentence case). I also added name for each settings. I also created new release here: https://github.com/bao-tg/kindle-vocab/releases/tag/1.0.3. 

Thank you so much for your constructive feedback sir, I've learned a lot.
